beacon-chain/operations/service.go

```diff
@@ -87,20 +98,31 @@ func (s *Service) saveOperations() {
 			log.Debug("Beacon block operations service context closed, exiting goroutine")
 			return
 		// Listen for a newly received incoming exit from the sync service.
-		case exit := <-s.incomingValidatorExit:
+		case exit := <-s.incomingValidatorExits:
 			hash, err := hashutil.HashProto(exit)
 			if err != nil {
 				log.Errorf("Could not hash exit req proto: %v", err)
 				continue
 			}
-
 			if err := s.beaconDB.SaveExit(exit); err != nil {
 				log.Errorf("Could not save exit request: %v", err)
-				s.errors = append(s.errors, err)
+				s.error = err
 				continue
 			}
-
 			log.Debugf("Exit request %#x saved in db", hash)
+		case attestation := <-s.incomingAtt:
+			hash, err := hashutil.HashProto(attestation)
+			if err != nil {
+				log.Errorf("Could not hash attestation proto: %v", err)
+				continue
+			}
+			if err := s.beaconDB.SaveAttestation(attestation); err != nil {
+				log.Errorf("Could not save attestation: %v", err)
+				s.error = err
```

Comment:
Do you think this is a good indicator of the node health?

The cluster would stop serving traffic to an unhealthy node and eventually kill it. This type of error seems to never recover so it might not be a good idea to set this permanent error state when a single attestation fails to save.

Reply:
I missed the fact cluster would stop serving traffic to unhealthy nodes. I think maybe a spectrum of health levels would be a good idea (ranging from warning -> unhealthy -> critical unhealthy..etc) The recoverable faulty attributes such as flakey internet connections and a one off failed DB save could fall under warning

I will remove `s.error = err` for now.